### DISCUSSION TITLE
Prevent overriding provided functions

### DISCUSSION THREAD

**Discussion by spaghetti2514** [josdejong/mathjs#2690] [Ideas]
![screen shot 2014-10-05 at 11 15 00 pm](https://cloud.githubusercontent.com/assets/1966861/4521391/7e5d1662-4d0f-11e4-822a-eb1d95705396.png)

**Comment by josdejong** [comment]
Thanks, we can create an option to prevent overriding built-in functions, constants, and units, to prevent accidental replacement. This should be optional though, because on the other side it can be very powerful to override functions, in order to extend them with new functionality.
